src/data/set/intervals/image_preimage.lean

```diff
@@ -20,6 +20,78 @@ lemmas about preimages and images of all intervals. We also prove a few lemmas a
 universe u
 
 namespace set
+
+section has_exists_add_of_le
+/-!
+The lemmas in this section state that addition maps intervals bijectively. The typeclass
+`has_exists_add_of_le` is defined specifically to make them work; it applies to all
+`ordered_add_comm_group`, but also to `ℕ` and `ℝ≥0`, which are not groups.
```

Comment:
```suggestion
`ordered_add_comm_group` and `canonically_ordered_add_monoid`s.
```

Reply:
Canonically ordered add monoids don't satisfy the claimed lemmas in general.

Reply:
Yes, good point, this one of my suggestions is wrong.